mne/viz/misc.py

```diff
@@ -894,3 +894,124 @@ def _handle_event_colors(unique_events, color, unique_events_id):
                 warn('Color is not available for event %d. Default colors '
                      'will be used.' % this_event)
     return color
+
+
+def plot_csd(csd, info=None, mode='csd', colorbar=True, cmap=None,
+             n_cols=None, show=True):
+    """Plot CSD matrices.
+
+    A sub-plot is created for each frequency. If an info object is passed to
+    the function, different channel types are plotted in different figures.
+
+    Parameters
+    ----------
+    csd : instance of CrossSpectralDensity
+        The CSD matrix to plot.
+    info: instance of Info | None
+        To split the figure by channel-type, provide the measurement info.
+        By default, the CSD matrix is plotted as a whole.
+    mode : 'csd' | 'coh'
+        Whether to plot the cross-spectral density ('csd', the default), or
+        the coherence ('coh') between the channels.
+    colorbar : bool
+        Whether to show a colorbar. Defaults to ``True``.
+    cmap : str | None
+        The matplotlib colormap to use. Defaults to None, which means the
+        colormap will default to matplotlib's default.
+    n_cols : int | None
+        CSD matrices are plotted in a grid. This parameter controls how
+        many matrix to plot side by side before starting a new row. By
+        default, a number will be chosen to make the grid as square as
+        possible.
+    show : bool
+        Whether to show the figure. Defaults to ``True``.
+
+    Returns
+    -------
+    fig : matplotlib figure | list of matplotlib figures
```

Comment:
maybe it should just always give a list, making the single-freq an exception often causes more difficulties than it's worth to save a `[0]` at the user end

Reply:
agreed